dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx

```diff
@@ -27,18 +28,28 @@ export const Gauge = ({ visualizations, layout, config }: any) => {
   // data config parametrs
   const { dataConfig = {}, layoutConfig = {} } = visualizations.data.userConfigs;
   const dataConfigTab = visualizations?.data?.rawVizData?.Gauge?.dataConfig;
-  const dimensions = dataConfigTab?.dimensions ? dataConfigTab?.dimensions : [];
-  const metrics = dataConfigTab?.metrics ? dataConfigTab?.metrics : [];
-  const dimensionsLength = dimensions.length && dimensions[0]?.name != '' ? dimensions.length : 0;
-  const metricsLength = metrics.length && metrics[0]?.name != '' ? metrics.length : 0;
+  const dimensions = dataConfigTab?.dimensions
+    ? dataConfigTab?.dimensions?.filter((i) => i.name !== '')
```

Comment:
You do not need these question marks here `dataConfigTab?.dimensions?.filter` This statement is executing, that means `dataConfigTab?.dimensions` has some values in it.